django/loader/load/loader.py

```diff
@@ -37,14 +37,23 @@ def load(self, fhir_instance, resource_type):
 
         if isinstance(resource, OperationOutcome):
             resource_id = get_resource_id(fhir_instance)
-            # Increment counter for failed validations
-            counter_failed_validations.labels(resource_id=resource_id, resource_type=resource_type).inc()
+
+            # A bit ugly, DuplicateError should have had code as a class attribute
+            if resource.issue[0].code == DuplicateError("").code:
+                message = "Document already present"
+            else:
+                # Increment counter for failed validations
+                counter_failed_validations.labels(resource_id=resource_id, resource_type=resource_type).inc()
+                message = "Validation failed"
+
             # Log
             logger.exception(
                 {
-                    "message": f"Validation failed\n"
-                    f"Diagnostics: {[issue.diagnostics for issue in resource.issue]}\n"
-                    f"Document: {fhir_instance}",
+                    "message": f"{message}",
+                    "diagnostics": "\n".join(issue.diagnostics for issue in resource.issue),
```

Comment:
use resource.format()

Reply:
`format` does the opposite: it builds a OperationOutcome from a list of error strings